File: src/plugins/paths.ts
import { homedir } from "node:os";
import { join } from "node:path";

/** User-level managed plugin root (`~/.rlm/plugins`). */
export function getUserPluginsRoot(): string {
  return join(homedir(), ".rlm", "plugins");
}

/** User-level installed plugin catalog (`~/.rlm/plugins/catalog.json`). */
export function getUserPluginCatalogPath(): string {
  return join(getUserPluginsRoot(), "catalog.json");
}

/** Install directory for a plugin id under the user catalog root. */
export function getUserPluginInstallDir(pluginId: string): string {
  return join(getUserPluginsRoot(), pluginId);
}

/** Project-local plugin catalog (`.rlm/plugins/catalog.json`). */
export function getProjectPluginCatalogPath(projectRoot: string): string {
  return join(projectRoot, ".rlm", "plugins", "catalog.json");
}

File: crates/rlm-core/src/plugins/paths.rs
use std::path::{Path, PathBuf};

pub fn user_plugins_root() -> PathBuf {
    dirs::home_dir()
        .map(|home| home.join(".rlm").join("plugins"))
        .unwrap_or_else(|| PathBuf::from(".rlm/plugins"))
}

pub fn user_plugin_catalog_path() -> PathBuf {
    user_plugins_root().join("catalog.json")
}

pub fn user_plugin_install_dir(plugin_id: &str) -> PathBuf {
    user_plugins_root().join(plugin_id)
}

pub fn project_plugin_catalog_path(project_root: &Path) -> PathBuf {
    project_root
        .join(".rlm")
        .join("plugins")
        .join("catalog.json")
}
